src/main.cpp

```diff
@@ -5757,7 +5791,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         } else if (status == READ_STATUS_FAILED) {
             // Might have collided, fall back to getdata now :(
             std::vector<CInv> invs;
-            invs.push_back(CInv(MSG_BLOCK, resp.blockhash));
+            invs.push_back(CInv(MSG_BLOCK | GetFetchFlags(pfrom, chainActive.Tip(), chainparams.GetConsensus()), resp.blockhash));
```

Comment:
I believe this is wrong. If a witness peer announces, directly to us, via compact blocks, the first segwit block, and we are missing the second-to-first segwit block (or its currently in-flight) when we receive the response to this message, we will ban the peer sending the first segwit block as it will respond in non-witness form, because of this request.